backend/courses/views.py

```diff
@@ -376,3 +378,43 @@ def get_queryset(self):
             section__course__semester=get_current_semester(),
             in_add_drop_period=True,
         ).order_by("created_at")
+
+
+# Function-Based View for Friendship Handling
+def sendFriendshipRequest(request):
+    res = {}
+    # verify that the friendship request can be sent (only if it doesn't already exist and
```

Comment:
Nice defensive mindset here - if we add the unique constraint above, we can just use `friendship, created = Friendship.get_or_create(...)` and return an error if `created` is false: 
https://docs.djangoproject.com/en/4.1/ref/models/querysets/#get-or-create